src/Raven.Server/ServerWide/Commands/Sharding/PutShardedSubscriptionCommand.cs

```diff
@@ -45,6 +46,14 @@ protected override void AssertValidChangeVector()
         // the CVs were validated in handler
     }
 
+    protected override void AssertCanPutSubscription(RawDatabaseRecord record)
+    {
+        if (record.IsSharded == false)
+        {
+            throw new PutSubscriptionException($"'{nameof(PutShardedSubscriptionCommand)}' is not supported for regular database.");
```

Comment:
Please make sure that this kind of exception will not cause the state machine to be stuck.

Reply:
It should be part of the excepted exception (or be wrapped in such)